src/Relations/BelongsToMany.php

```diff
@@ -229,6 +230,8 @@ public function detach($ids = [], $touch = true)
         }
 
         // Remove the relation to the parent.
+        assert($this->parent instanceof \MongoDB\Laravel\Eloquent\Model);
+        assert($query instanceof \MongoDB\Laravel\Eloquent\Model);
```

Comment:
```suggestion
        assert($query instanceof \MongoDB\Laravel\Eloquent\Builder);
```
The `$query` variable is an instance of `Builder` class.